arch/cortex-m4/src/scb.rs

```diff
@@ -33,6 +33,22 @@ const SCB_BASE: usize = 0xE000ED00;
 
 static mut SCB: *mut ScbRegisters = SCB_BASE as *mut ScbRegisters;
 
+/// Allow the core to go into deep sleep on WFI.
+///
+/// The specific definition of "deep sleep" is chip specific.
+pub unsafe fn set_sleepdeep() {
```

Comment:
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0497a/CHDJJHJI.html
> If your device does not support two sleep modes, the effect of changing the value of this bit is implementation-defined.

While I suspect in practice this would resolve to a no-op, would it make more sense architecturally to have the implementation in the chip?